h2o-r/demos/rdemo.airlines.target_encode_new.R

```diff
@@ -0,0 +1,164 @@
+full_train <- h2o.rbind(train, te_holdout)
+default_gbm <- h2o.gbm(x = myX, y = "IsDepDelayed",
+                       training_frame = full_train, validation_frame = valid,
+                       ntrees = 1000, score_tree_interval = 10, model_id = "default_gbm",
+                       # Early Stopping
+                       stopping_rounds = 5, stopping_metric = "AUC", stopping_tolerance = 0.001,
+                       seed = 1)
+
+
+
+############################################## LeaveOneOut #############################################################
+print("Perform Leave One Out Target Encoding on cabin, embarked, and home.dest")
+
+te_cols <- list("Origin", "Dest")
+# For this model we will calculate LOO Target Encoding on the full train
+# There is possible data leakage since we are creating the encoding map on the training and applying it to the training
+# To mitigate the effect of data leakage without creating a holdout data, we remove the existing value of the row (holdout_type = LeaveOneOut)
+
+loo_train <- full_train
+loo_valid <- valid
+loo_test <- test
+
+# Create Leave One Out Encoding Map
+encoding_map <- h2o.target_encode_fit(full_train, te_cols, "IsDepDelayed")
+
+
+# Apply Leave One Out Encoding Map on Training, Validation, Testing Data
+loo_train <- h2o.target_encode_transform(frame = loo_train, x = te_cols, y = "IsDepDelayed",
+                                        target_encode_map = encoding_map, holdout_type = "loo",
+                                        blended_avg = TRUE, inflection_point = 10, smoothing = 5,
+                                        noise = -1, seed = 1234)
+
+loo_valid <- h2o.target_encode_transform(frame = loo_valid, x = te_cols, y = "IsDepDelayed",
+                                        target_encode_map = encoding_map, holdout_type = "none",
+                                        blended_avg = TRUE, inflection_point = 10, smoothing = 5,
+                                        noise = 0, seed = 1234)
+
+loo_test <- h2o.target_encode_transform(frame = loo_test, x = te_cols, y = "IsDepDelayed",
+                                        target_encode_map = encoding_map, holdout_type = "none",
+                                        blended_avg = TRUE, inflection_point = 10, smoothing = 5,
+                                        noise = 0, seed = 1234)
+
+print("Run GBM with Leave One Out Target Encoding")
+myX <- setdiff(colnames(loo_test), c(te_cols, "IsDepDelayed", "IsDepDelayed_REC"))
+loo_gbm <- h2o.gbm(x = myX, y = "IsDepDelayed",
+                   training_frame = loo_train, validation_frame = loo_valid,
+                   ntrees = 1000, score_tree_interval = 10, model_id = "loo_gbm",
+                   # Early Stopping
+                   stopping_rounds = 5, stopping_metric = "AUC", stopping_tolerance = 0.001,
+                   seed = 1)
+
+h2o.varimp_plot(loo_gbm)
+
```

Comment:
It would be very useful to see the AUC on the default model vs target encoded model for each of the types of target encoding.  Could you add a line of code that computes AUC on the test set? 

```
 h2o.auc(h2o.performance(loo_gbm, loo_test))
```

We would hope that the target encoded models should have better performance (we want to verify this).

Reply:
@ledell I believe that I have these comparisons in both benchmark files `rdemo.airlines.target_encode_new.R` and `rdemo.titanic.target_encode_new.R`. Those are in the ends of the files.

```
print("Compare AUC of GBM with different types of target encoding")

print(paste0("Default GBM AUC: ", round(h2o.auc(h2o.performance(default_gbm, test)), 3)))

print(paste0("LOO GBM AUC: ", round(h2o.auc(h2o.performance(loo_gbm, loo_test)), 3)))

print(paste0("KFold TE GBM AUC: ", round(h2o.auc(h2o.performance(kfold_te_gbm, kfold_test)), 3)))

print(paste0("Holdout GBM AUC: ", round(h2o.auc(h2o.performance(holdout_gbm, holdout_test)), 3)))
```